bot/src/main/kotlin/org/epilink/bot/rulebook/QuickHttp.kt

```diff
@@ -8,25 +8,94 @@
  */
 package org.epilink.bot.rulebook
 
+import com.fasterxml.jackson.core.type.TypeReference
 import com.fasterxml.jackson.databind.ObjectMapper
-import com.fasterxml.jackson.module.kotlin.readValue
+import com.fasterxml.jackson.module.kotlin.jacksonTypeRef
 import io.ktor.client.HttpClient
 import io.ktor.client.engine.apache.Apache
-import io.ktor.client.features.auth.Auth
-import io.ktor.client.features.auth.providers.basic
 import io.ktor.client.request.get
 import io.ktor.client.request.header
 import io.ktor.http.ContentType
 import io.ktor.http.HttpHeaders
+import io.ktor.utils.io.core.*
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 import org.slf4j.LoggerFactory
+import java.util.*
 
 /*
  * Utility functions for use within a Rulebook
  */
 
 private val logger = LoggerFactory.getLogger("epilink.rulebooks.helpers")
+private val client = HttpClient(Apache)
+
+/**
+ * An authentication method for [httpGetJson] requests
+ */
+sealed class Auth
+
+/**
+ * Basic HTTP authentication method for [httpGetJson] requests
+ *
+ * @param username The username to provide
+ * @param password The password to provide
+ */
+data class Basic(val username: String, val password: String) : Auth()
+
+/**
+ * Bearer token authentication method for [httpGetJson] requests
+ *
+ * @param token The token to provide
+ */
+data class Bearer(val token: String) : Auth()
+
+/**
+ * Perform a HTTP GET request to the given URL, expecting a JSON reply, and return this JSON reply deserialized to the
+ * inferred return type.
```

Comment:
```suggestion
 * given return type.
```

Types are not always inferred, especially in this case